webui/react/src/components/ExperimentActionDropdown.tsx

```diff
@@ -83,9 +84,25 @@ const ExperimentActionDropdown: React.FC<Props> = ({
   const ExperimentEditModal = useModal(ExperimentEditModalComponent);
   const ExperimentMoveModal = useModal(ExperimentMoveModalComponent);
   const HyperparameterSearchModal = useModal(HyperparameterSearchModalComponent);
+  const [experimentItem, setExperimentItem] = useState<ExperimentItem | undefined>(undefined);
```

Comment:
mmph, i'm not a fan of not having any user feedback for the load state

i think ideally we'd fix the underlying issue where modals run stuff before they're open, but as an intermediate fix, can we introduce some sort of interstitial modal?

like, imagine the following code:

```tsx
const [data, setData] = useState<Loadable<Data>>(NotLoaded);
const InterstitialModal = useModal(InterstitialModalComponent);
const LoadedDataModal = useModal(LoadedDataComponent);
//snip
return (
<div>
  {/* snip */}
  {data.map((d) => (<LoadedDataModal.Component data={data} />)).getOrElse(null)}
  <InterstitialModal.Component close={InterstitialModal.close} loadable={data} then={LoadedDataModal.open} />
</div>
)
```

the interstititial modal would show a spinner until the loadable element is loaded, at which point it closes itself (by calling the `close` prop) and then doing whatever's in the `then` prop

Reply:
I was thinking to add a loading indicator in menu after the button is clicked. I'll look into it more

Reply:
@ashtonG I made it like this. what do you think? 

https://github.com/determined-ai/determined/assets/109113616/4f0c2340-c60a-4528-8d6e-4983bfd088b0

Reply:
selecting other menu items while the experiment is loading can cause modals to clash and the loading state implies that the user can cancel the opening by closing the menu but they can't.

this doesn't feel very responsive.

Reply:
ok, i made `InterstitialModal` as you suggested. 
What do you think?

https://github.com/determined-ai/determined/assets/109113616/1df3a25e-ce23-4d1e-aae9-226a7409b9d3

Reply:
that's a good point -- i noticed you implemented the close reasons as well. What we could do is instead of having a `then` prop that is called when the `loadable` is loaded, we could instead use the reason passed to the `close` handler to determine whether or not to continue on to the next modal

Reply:
i don't quite get it. Could you elaborate on this?

Reply:
so, in the interstitial modal, we have that effect that checks whether or not the `data` prop is loaded:
```tsx
    if (loadableData.isLoaded) {
      close('ok');
      then();
    } else if (loadableData.isFailed) {
      close('failed');
    }
```
given that we're already calling `close` with the reason why it's closing, we should be able to change the `close` prop we're passing to the interstitial. Instead of directly calling the modal close, we can instead check the reason, and if it's `'ok'`, we open up the next modal.

Reply:
ok if i understand it correctly, this should be fixed.